dags/sources/tasks/business_logic/source_data_normalize.py

```diff
@@ -47,18 +47,180 @@ def source_data_normalize(
     df = df_acteur_from_source
 
     # Renommage des colonnes
+    # DEPRECATED: le rename est désormais géré comme une transformation
     df = df.rename(columns={k: v for k, v in column_mapping.items() if v is not None})
 
-    # Transformation des colonnes
-    # Appliquer la transformation à la colonne d'origine et stocker le résultat
-    # dans la colonne de destination
-    for transformation in column_transformations:
-        function_name = transformation["transformation"]
-        df[transformation["destination"]] = df[transformation["origin"]].apply(
-            TRANSFORMATION_MAPPING[function_name]
+    ### TRANSFORMATIONS
+    # TODO : mettre cette doc dans la doc technique si accepté par les collègues
+    #
+    # Plusieurs types de transformations à executer dans l'ordre suivant
+    # 1. Renommage des colonnes
+    #
+    # { "origin": "col origin", "destination": "col origin" }
+    #
+    # 2. Transformation des colonnes
+    #
+    # { "origin": "col origin", "destination": "col destination",
+    #   "transformation": "function_name" }
+    #
+    # 3. Transformation du dataframe
+    #
+    # { "origin": ["col origin 1", "col origin 2"], "transformation": "function_name",
+    #   "destination": ["col destination 1", "col destination 2"] }
+    #
+    # 4. Ajout des colonnes avec une valeur par défaut
+    #
+    # { "colonne": "col 1", "value" : "val" }
+    #
+    # 5. Supression des colonnes
+    #
+    # { "remove": "col 1" }
+    #
+    # 6. Colonnes à garder (rien à faire, utilisé pour le controle)
+    #
+    # { "keep": "col 1" }
+    #
+    # Les transformations sont définies dans le fichier de configuration
+    # Après que toutes les transformations aient été appliquées, on vérifie que
+    # la dataframe a exactement les colonne attendues
+
+    # 1. Renommage des colonnes
+    #
+    # { "origin": "col origin", "destination": "col origin" }
+    #
+    # J'ai une origine de type string et une destination de type string et pas de
+    # transformation
+
+    columns_to_rename = [
+        t
+        for t in column_transformations
+        if "origin" in t
+        and isinstance(t["origin"], str)
+        and "destination" in t
+        and isinstance(t["destination"], str)
+        and "transformation" not in t
+    ]
+    df = df.rename(
+        columns={
+            column_to_rename["origin"]: column_to_rename["destination"]
+            for column_to_rename in columns_to_rename
+        }
+    )
```

Comment:
J'ai tendance à tout tronçonner, et aurais été tenté d'extraire ça dans une petite fonction, genre 

```suggestion
def rename_columns_from(column_transformations, df) -> pd.DataFrame:
     columns_to_rename = [
         t
         for t in column_transformations
         if "origin" in t
         and isinstance(t["origin"], str)
         and "destination" in t
         and isinstance(t["destination"], str)
         and "transformation" not in t
     ]
     return df.rename(
         columns={
             column_to_rename["origin"]: column_to_rename["destination"]
             for column_to_rename in columns_to_rename
         }
     )


def source_data_normalize():
    # ... 
    df_with_renamed_columns = rename_columns_from(column_transformations, df)
    df_with_transformed_columns = transform_columns_from(column_transformations, df_with_renamed_clumns)
    df_transformed = transform_df_from(columns_transformations, df_with_transformed_columns)
    df_transformed_with_default = set_default_column_from(columns_transformations, df_transformed_
    # etc etc 
```

C'est vraiment une question de gout / couleur, mais je trouve que ça apporte de la lisibilité, permet presque de se passer de commentaire, ou alors de les faire remonter en docstring dans chaque fonction individuelle et ça s'auto documentera un jour

Reply:
why not !

Mais alors en fonction privé `_rename_columns_from` ?

Reply:
ouais pourquoi pas ! (pas trop d'avis haha)

Reply:
done !